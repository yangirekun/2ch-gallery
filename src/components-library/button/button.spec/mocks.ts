export const mockProps = [
  {
    id: "button-test-id",
    label: "button-test-text",
    className: "button-test-class",
    disabled: false,
    onClick: jest.fn(),
  },
  {
    id: "button-test-id",
    label: "button-test-text",
    className: "button-test-class",
    disabled: true,
    onClick: jest.fn(),
  },
  {
    id: "button-test-id",
  },
];
