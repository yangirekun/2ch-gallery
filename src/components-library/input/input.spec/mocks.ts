export const mockProps = [
  {
    id: "input-test-id",
    label: "input-test-label",
    className: "input-test-class",
    placeholder: "input-test-placeholder",
    value: "input-test-value",
    onChange: jest.fn(),
  },
];
