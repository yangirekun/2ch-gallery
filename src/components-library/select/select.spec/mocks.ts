const mockList = [
  { id: "b", label: "Бред" },
  { id: "vg", label: "Video Games General" },
  { id: "po", label: "Политика" },
  { id: "fag", label: "Фагготрия" },
  { id: "news", label: "Новости" },
  { id: "sex", label: "Секс и отношения" },
  { id: "ma", label: "Манга" },
  { id: "hw", label: "Компьютерное железо" },
  { id: "biz", label: "Бизнес" },
  { id: "v", label: "Video Games" },
  { id: "wrk", label: "Работа и карьера" },
  { id: "ftb", label: "Футбол" },
  { id: "pr", label: "Программирование" },
  { id: "cc", label: "Криптовалюта" },
  { id: "rf", label: "Убежище" },
  { id: "au", label: "Автомобили" },
  { id: "alco", label: "Алкоголь" },
  { id: "2d", label: "Аниме/Беседка" },
  { id: "mov", label: "Фильмы" },
  { id: "ja", label: "Японская культура" },
  { id: "soc", label: "Общение" },
  { id: "gg", label: "Хорошие девушки" },
  { id: "mobi", label: "Мобильные устройства и приложения" },
  { id: "me", label: "Медицина" },
  { id: "sp", label: "Спорт" },
  { id: "cg", label: "Консоли" },
  { id: "bo", label: "Книги" },
  { id: "gsg", label: "Grand Strategy Games" },
  { id: "a", label: "Аниме" },
  { id: "bi", label: "Велосипеды" },
  { id: "wm", label: "Военная техника" },
  { id: "fa", label: "Мода и стиль" },
  { id: "fiz", label: "Физкультура" },
  { id: "by", label: "Беларусь" },
  { id: "mo", label: "Мотоциклы" },
  { id: "hi", label: "История" },
  { id: "psy", label: "Психология и психиатрия" },
  { id: "w", label: "Оружие" },
  { id: "brg", label: "Браузерные игры" },
  { id: "tes", label: "The Elder Scrolls" },
  { id: "bg", label: "Настольные игры" },
  { id: "hry", label: "Highlights of Russian-speaking youtube" },
  { id: "gd", label: "Gamedev" },
  { id: "s", label: "Программы" },
  { id: "pa", label: "Живопись" },
  { id: "fd", label: "Фэндом" },
  { id: "d", label: "Скла/d/ грязи" },
  { id: "fs", label: "Фэнтези" },
  { id: "fet", label: "Фетиш" },
  { id: "em", label: "Другие страны" },
  { id: "dr", label: "Дневнички" },
  { id: "re", label: "Религия" },
  { id: "spc", label: "Космос и астрономия" },
  { id: "diy", label: "Хобби" },
  { id: "mus", label: "Музыканты" },
  { id: "socionics", label: "Соционика и психософия" },
  { id: "c", label: "Комиксы и мультфильмы" },
  { id: "un", label: "Образование" },
  { id: "p", label: "Фотография" },
  { id: "fg", label: "Трапы" },
  { id: "e", label: "Extreme Pron" },
  { id: "di", label: "Столовая" },
  { id: "td", label: "Трёхмерная графика" },
  { id: "kpop", label: "K-pop" },
  { id: "obr", label: "Offline Battle Rap" },
  { id: "tv", label: "Сериалы" },
  { id: "hh", label: "Hip-Hop" },
  { id: "ra", label: "Радиотехника" },
  { id: "sn", label: "Паранормальные явления" },
  { id: "wh", label: "Warhammer" },
  { id: "dom", label: "Домострой" },
  { id: "fur", label: "Фурри" },
  { id: "ga", label: "Геи" },
  { id: "mu", label: "Музыка" },
  { id: "h", label: "Хентай" },
  { id: "hc", label: "Hardcore" },
  { id: "hv", label: "Халява в интернете" },
  { id: "sci", label: "Наука" },
  { id: "ho", label: "Прочий хентай" },
  { id: "zog", label: "Теории заговора" },
  { id: "fl", label: "Иностранные языки" },
  { id: "mmo", label: "Massive multiplayer online games" },
  { id: "mg", label: "Магия" },
  { id: "t", label: "Техника" },
  { id: "izd", label: "Графомания" },
  { id: "vn", label: "Визуальные новеллы" },
  { id: "law", label: "Право и правоприменители" },
  { id: "qtr4", label: "qtr4" },
  { id: "asmr", label: "ASMR" },
  { id: "vape", label: "Электронные сигареты" },
  { id: "ruvn", label: "Российские визуальные новеллы" },
  { id: "ph", label: "Философия" },
  { id: "wr", label: "Текстовые авторские рпг" },
  { id: "rm", label: "Rozen Maiden" },
  { id: "ukr", label: "/Ukr/aine — Україна" },
  { id: "wow", label: "World of Warcraft" },
  { id: "es", label: "Бесконечное Лето" },
  { id: "whn", label: "Warhammer Nihilus" },
  { id: "ne", label: "Животные и природа" },
  { id: "de", label: "Дизайн" },
  { id: "aa", label: "Аниме арт" },
  { id: "web", label: "Web-мастера" },
  { id: "out", label: "Активный отдых" },
  { id: "media", label: "Анимация" },
  { id: "m", label: "Меха" },
  { id: "se", label: "Самообразование" },
  { id: "kz", label: "Казачан" },
  { id: "o", label: "Рисовач" },
  { id: "ch", label: "Чатики и конфочки" },
  { id: "math", label: "Математика" },
  { id: "ew", label: "Конец света" },
  { id: "mlp", label: "My Little Pony" },
  { id: "pvc", label: "Фигурки" },
  { id: "sf", label: "Научная фантастика" },
  { id: "tr", label: "Транспорт и авиация" },
  { id: "wp", label: "Обои и высокое разрешение" },
  { id: "mc", label: "Minecraft" },
  { id: "to", label: "Touhou" },
  { id: "r", label: "Просьбы" },
  { id: "br", label: "Барахолка" },
  { id: "trv", label: "Путешествия и отдых" },
  { id: "gb", label: "Азартные игры" },
  { id: "moba", label: "Multiplayer Online Battle Arena games" },
  { id: "abu", label: "abu" },
  { id: "cul", label: "Культура" },
  { id: "old", label: "Олдфаги" },
  { id: "guro", label: "Шок-контент" },
  { id: "ussr", label: "СССР" },
  { id: "jsf", label: "Japanese Street Fashion" },
  { id: "sw", label: "Star Wars" },
  { id: "ya", label: "Яой" },
  { id: "r34", label: "Правило 34" },
  { id: "gabe", label: "Gabe Logan Newell" },
  { id: "cute", label: "Милое" },
  { id: "8", label: "8bit" },
  { id: "mlpr", label: "My Little Pony Roleplay" },
  { id: "ro", label: "Рогалики" },
  { id: "who", label: "Doctor Who" },
  { id: "srv", label: "Серверы" },
  { id: "electrach", label: "Электронная музыка" },
  { id: "ing", label: "Инженерное дело" },
  { id: "got", label: "Песнь льда и пламени" },
  { id: "crypt", label: "Криптоанархия" },
  { id: "lap", label: "Лапник (4x4 клуб)" },
  { id: "smo", label: "Курение" },
  { id: "hg", label: "Хентайные игры/hentai games" },
  { id: "sad", label: "Садоводство" },
  { id: "fi", label: "Рыбалка" },
  { id: "nvr", label: "Новороссия: ЛНР, ДНР" },
  { id: "ind", label: "Инди" },
  { id: "ld", label: "Осознанные сновидения" },
  { id: "fem", label: "Феминизм" },
  { id: "vr", label: "Виртуальная реальность" },
  { id: "arg", label: "ARG/игры в альтернативной реальности" },
  { id: "char", label: "Сетевые персонажи" },
  { id: "pok", label: "Pokemon" },
  { id: "wwe", label: "WorldWide Wrestling Universe" },
  { id: "int", label: "International" },
  { id: "catalog", label: "Статьи" },
  { id: "api", label: "API" },
];

export const mockProps = [
  {
    id: "select-test-id",
    label: "select-test-label",
    className: "select-test-class",
    placeholder: "select-test-placeholder",
    list: mockList,
    value: mockList[0].id,
    onChange: jest.fn(),
  },
];
