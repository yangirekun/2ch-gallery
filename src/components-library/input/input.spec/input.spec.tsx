import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Input } from "../index";
import { mockProps } from "./mocks";

describe("testing <Input />", () => {
  test("it should be rendered correctly", () => {
    expect(render(<Input {...mockProps[0]} />).asFragment()).toMatchSnapshot();
  });

  test("it should call onChange on user input", () => {
    const { getByTestId } = render(<Input {...mockProps[0]} />);
    const input: any = getByTestId("input-test-id-input");
    fireEvent.change(input, { target: { value: "Значение" } });
    expect(mockProps[0].onChange).toBeCalledTimes(1);
  });
});
