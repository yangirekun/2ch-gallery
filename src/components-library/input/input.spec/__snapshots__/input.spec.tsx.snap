// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Input /> it should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="input input-test-class"
    data-testid="input-test-id-wrapper"
  >
    <label
      class="input__label"
      data-testid="input-test-id-label"
      for="input-test-id"
    >
      input-test-label
    </label>
    <input
      class="input__field"
      data-testid="input-test-id-input"
      id="input-test-id"
      placeholder="input-test-placeholder"
      type="text"
      value="input-test-value"
    />
  </div>
</DocumentFragment>
`;
